### ISSUE TITLE
allow to apply the AuthorProcessor to others tags

### ISSUE THREAD

**Issue by renanbr** [renanbr/bibtex-parser#25]
When enabled, the `AuthorProcessor` modifies only the `author` tag.

It could be interesting to apply the same processor in others tags, for example `editor`.

**Comment by renanbr** [comment]
Tags don't vary much, then `AuthorProcessor` could be superseded by `NameProcessor`, which:

- by default it process `author` and `editor` tags;
- its constructor accepts an optional argument for overriding the default tags.

**Comment by andrechalom** [comment]
From issues #25 and #26, both `NameProcessor` and `LatexToUnicodeProcessor` should have an optional argument for setting the tags to which they should be applied, right?

I think this can also include something like a wildcard setting, for applying the processor to all tags.

Maybe it's best to implement a base class `RenanBr\BibTexParser\Processor\Processor` from which these classes will inherit the behavior related to selecting the tags? This base class should include a `protected $apply_to` which might be overriden, so it will also be used by `KeywordProcessor`.

**Comment by renanbr** [comment]
> I think this can also include something like a wildcard setting, for applying the processor to all tags.

I agree. What about `setTagCoverage(array $tags, string $strategy = null)` in which $tags is a list of tags' names; and $strategy may assume "whitelist" (default) or "blacklist"?

> Maybe it's best to implement a base class (...)

Why not a [trait](https://secure.php.net/manual/en/language.oop5.traits.php)?

```php
trait RenanBr\BibTexParser\Processor\TagCoverageTrait
{
    // processors can setup the default behavior AND final users can change it
    public function setTagCoverage(array $tags, string $strategy = null): void;

    // processors can, internally, use it before applying modifications
    protected function isTagCovered(string $tag): bool;
}
```

Then we can keep constructors clean; reuse this approach for all current and future processors; and provide the API for custom implementations as well.

**Comment by andrechalom** [comment]
Is this "tag coverage" behavior going to be optional for a Processor?

If yes, I think implementing the trait as you suggested is a great
solution.

But if we consider that all Processors (premade and custom) must accept a
call to `setTagCoverage`, I think simple inheritance provides a cleaner
alternative. It's a bit of uniformization, although it may be a somewhat
silly uniformization - after all, we can force custom Processors to accept
the calls to `setTagCoverage`, but we can't force them to honor the
coverage set...

TL;DR: I see good arguments for either class or trait :P
